mypyc/namegen.py

```diff
@@ -36,9 +40,7 @@ class NameGenerator:
       prefixes will be 'bar_foo_' and 'baz_foo_'.
 
     * Replace '.' in the Python name with '___' in the C name. (And
-      replace the unlikely but possible '___' with '___3_'. This
-      collides '___' with '.3_', but this is OK because names
-      may not start with a digit.)
+      replace '___' with '____' and so on to prevent collisions)
```

Comment:
This doesn't seem to help with the unicode collisions, though?

Reply:
Oh, and it introduces new collisions. `a._` and `a___` (3 underscores) now both map to `a____` (4 underscores), for example.

Reply:
Oops!